### PR TITLE
Post: quarkus.search.io series - rolling over

### PR DESCRIPTION
**Pull request by marko-bekhta**
@yrodiere here it is 😃 🙈

### CODE REVIEW

**Review by gsmet — COMMENTED** [review]
Looks very good overall, I added a few suggestions.

**Comment by marko-bekhta** [comment]
Thanks Guillaume! I've tried to address your suggestions.

File: _posts/2024-04-30-search-indexing-rollover.adoc
Comment (yrodiere) [inline _posts/2024-04-30-search-indexing-rollover.adoc:21]: This is a good intro, but quite restrictive as not many readers of the Quarkus blog will be in this position. They may stop reading right here.

Why not something like:

```suggestion
Does your application need full-text search capabilities? Do you need to keep your application running and producing search results without any downtime, even when reindexing all your data?
Look no further. In this post, we'll cover how you can approach this problem
and solve it in practice with a few low-level APIs, provided you use Hibernate Search, be it link:{quarkus-hibernate-search-docs-url}[on top of Hibernate ORM] or https://quarkus.io/version/main/guides/hibernate-search-standalone-elasticsearch[in standalone mode].
```

File: _posts/2024-04-30-search-indexing-rollover.adoc
Comment (yrodiere) [inline _posts/2024-04-30-search-indexing-rollover.adoc:23]:
```suggestion
The approach suggested in this post is based on the fact that Hibernate Search uses
```

File: _posts/2024-04-30-search-indexing-rollover.adoc
Comment (yrodiere) [inline _posts/2024-04-30-search-indexing-rollover.adoc:27]:
```suggestion
while an indexing operation will be sent to a write index alias.
```

File: _posts/2024-04-30-search-indexing-rollover.adoc
Comment (yrodiere) [inline _posts/2024-04-30-search-indexing-rollover.adoc:39]: This is a bit complex, I'd separate the paragraphs to clarify the two ideas are separate.

```suggestion
as the data on which the index documents are based is modified, providing a near real-time index synchronisation.

On the other hand, if the search requirements allow for a delay in synchronisation
```

File: _posts/2024-04-30-search-indexing-rollover.adoc
Comment (yrodiere) [inline _posts/2024-04-30-search-indexing-rollover.adoc:48]:
```suggestion
Now, since we want our application to keep  providing results to any search requests while we add/update documents to the indexes,
```

File: _posts/2024-04-30-search-indexing-rollover.adoc
Comment (yrodiere) [inline _posts/2024-04-30-search-indexing-rollover.adoc:49]: You're already stating this involves a purge further down.

```suggestion
we cannot perform a simple reindexing operation
```

File: _posts/2024-04-30-search-indexing-rollover.adoc
Comment (yrodiere) [inline _posts/2024-04-30-search-indexing-rollover.adoc:51]:
```suggestion
or the recently added link:{quarkus-hibernate-search-docs-url}#management[management endpoint in Quarkus],
```

File: _posts/2024-04-30-search-indexing-rollover.adoc
Comment (yrodiere) [inline _posts/2024-04-30-search-indexing-rollover.adoc:52]:
```suggestion
as these would drop all existing documents from the index before indexing them: search operations would not be able to match them anymore until reindexing finishes.
```

File: _posts/2024-04-30-search-indexing-rollover.adoc
Comment (yrodiere) [inline _posts/2024-04-30-search-indexing-rollover.adoc:58]:
```suggestion
Since Hibernate Search does not provide the rollover feature out of the box (https://hibernate.atlassian.net/browse/HSEARCH-3499[yet])
```

File: _posts/2024-04-30-search-indexing-rollover.adoc
Comment (yrodiere) [inline _posts/2024-04-30-search-indexing-rollover.adoc:99]:
```suggestion
```

File: _posts/2024-04-30-search-indexing-rollover.adoc
Comment (yrodiere) [inline _posts/2024-04-30-search-indexing-rollover.adoc:76]:
```suggestion
    });
```

File: _posts/2024-04-30-search-indexing-rollover.adoc
Comment (yrodiere) [inline _posts/2024-04-30-search-indexing-rollover.adoc:73]:
```suggestion
        var createIndexRequestBody = export.extension(ElasticsearchExtension.get())
                .bodyParts().get(0); // <4>
```

File: _posts/2024-04-30-search-indexing-rollover.adoc
Comment (yrodiere) [inline _posts/2024-04-30-search-indexing-rollover.adoc:135]:
```suggestion
except for the write alias (which is implicitly updated since the rollover request is targeting it directly).
```

File: _posts/2024-04-30-search-indexing-rollover.adoc
Comment (yrodiere) [inline _posts/2024-04-30-search-indexing-rollover.adoc:140]: It's probably worth reminding the reader what this was supposed to do?

```suggestion
With this successfully completed, indexes are in the state we wanted:

image::write-app.png[]

We can start populating our write index without affecting search requests.

Once we are done with indexing,
```

File: _posts/2024-04-30-search-indexing-rollover.adoc
Comment (yrodiere) [inline _posts/2024-04-30-search-indexing-rollover.adoc:277]:
```suggestion
feel free to reach out to us on the https://hibernate.atlassian.net/browse/HSEARCH-3499[pending feature requests] to discuss your ideas and suggestions.
```

**Review by yrodiere — APPROVED** [review]
Looks great! I'm curious: what did you use for the drawings?

I added a few suggestions below, feel free to pick what you agree with.

On important thing though: the feature request for rollover already exists, it's https://hibernate.atlassian.net/browse/HSEARCH-3499 . I added suggestions accordingly.

**Comment by marko-bekhta** [comment]
Yoann, thanks for the suggestions! I've applied them all (at least I think I did 😄)

as for the drawing tool, I was trying out this one https://excalidraw.com/

**Comment by marko-bekhta** [comment]
I'll let you decide on the date when you'd like to publish this one 😃

**Comment by yrodiere** [comment]
Thanks. Well then, seems we're ready to merge? We'll just need to agree on a date so that @marko-bekhta can update the post's metadata before we merge. @gsmet is there some process or is anyone free to post whenever they like?

**Comment by gsmet** [comment]
You can post whenever you like. Thursday is OK. I don't recommend Monday or Friday. I was told around posting around 3PM was the best (if you plan on tweeting about it).

I can either retweet or tweet (if you write something for me) with the QuarkusIO account if you'd like. But please ping me on Zulip when the blog post is published so that I don't have to monitor things :).

**Comment by yrodiere** [comment]
@marko-bekhta today 3PM then?

**Comment by gsmet** [comment]
Obviously you can push the blog post earlier. 3PM is more about when we should tweet.

**Comment by marko-bekhta** [comment]
> @marko-bekhta today 3PM then?

Updated the file name and the date.
Or should I use the date with time? Not sure how that works 😃

**Comment by yrodiere** [comment]
Merged, thanks. @marko-bekhta I'll let you tweet about this and/or provide Guillaume with a tweet for the `@quarkusio` account on X.

**Pull request merged**
